### PR TITLE
Support reading delta tables

### LINKED ISSUE #176: Add support for reading Delta tables
#### Describe your feature request

The Delta format is gaining popularity. The Polars project has recently added _read_delta methods for the Python and Rust APIs. It would be great if the node implementation could also support the Delta format.

I am not aware of any native Delta kernel implementation for Node, but perhaps a first version for nodejs-polars could use the Rust implementation behind-the-scenes?

### PR DESCRIPTION
**Pull request by Bidek56**
Adding support for reading delta tables thru Rust [deltalake](https://github.com/delta-io/delta-rs)
To close #176

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -27,6 +27,7 @@ thiserror = "1"
 smartstring = { version = "1" }
 serde_json = { version = "1" }
 either = "1.9"
+deltalake = "0.17.1"
```
Comment (universalmind303) [inline Cargo.toml:30]: I'm a bit hesitant to add this in as a dependency. 

this pulls in the entire arrow-rs project as well as many other heavy dependencies.

Comment (Bidek56) [reply Cargo.toml:30]: It makes sense, I will keep looking for alternatives. Thx

Comment (universalmind303) [reply Cargo.toml:30]: FWIW, I think the "ideal" approach would be first to upstream some work to delta-rs and expose a javascript library. Then we could use the JS library directly like py-polars does, and do a bunch of the logic on the JS side using `apache-arrow` as an optional dependency. 


If the `apache-arrow` npm package becomes a point of contention, we could likely bypass it by dynamically linking to the delta js bindings directly.

Comment (Bidek56) [reply Cargo.toml:30]: `delta-rs` team is busy with C# implementation now. :)

Comment (Bidek56) [reply Cargo.toml:30]: There was effort to use [delta-sharing](https://github.com/delta-incubator/delta-sharing-connectors) using JS and [Danfo.js](https://danfo.jsdata.org/).

**Pull request closed without merging**